storage/optimistic_kv_store.py

```diff
@@ -0,0 +1,103 @@
+from collections import OrderedDict
+from typing import Tuple, List, Iterable
+
+from storage.kv_store import KeyValueStorage
+
+
+class OptimisticKVStore:
```

Comment:
How are we going to use it?

Reply:
This will be used as base class of `IdrCache` in Indy-node